test/systemFlow_test.go

```diff
@@ -0,0 +1,374 @@
+package test
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+
+	server "github.com/curio-research/keystone/server"
+	"github.com/curio-research/keystone/state"
+	pb_test "github.com/curio-research/keystone/test/proto/pb.test"
+	"github.com/curio-research/keystone/test/testutils"
+	"github.com/gorilla/websocket"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var BookTable = state.NewTableAccessor[Book]()
+
+var p *testutils.PortManager
+
+func init() {
+	p = testutils.NewPortManager()
+}
+
+func TestAddBook(t *testing.T) {
+	e, ws, s, _ := startServer(t)
+	defer tearDown(ws, s)
+
+	w := e.World
+
+	playerID := 7
+	player2ID := 8
+
+	specificBookEntity := 969
+	err := sendWSMsg(ws, playerID, &pb_test.TestBookInfo{
+		Op:     pb_test.Operation_AddSpecific,
+		Title:  testBookTitle1,
+		Author: testBookAuthor1,
+		Entity: int64(specificBookEntity),
+	})
+	require.Nil(t, err)
+
+	err = sendWSMsg(ws, playerID, &pb_test.TestBookInfo{
+		Op:     pb_test.Operation_Add,
+		Title:  testBookTitle2,
+		Author: testBookAuthor2,
+	})
+	require.Nil(t, err)
+
+	err = sendWSMsg(ws, player2ID, &pb_test.TestBookInfo{
+		Op:     pb_test.Operation_Add,
+		Title:  testBookTitle3,
+		Author: testBookAuthor3,
+	})
+	require.Nil(t, err)
+
+	b1 := BookTable.Get(w, specificBookEntity)
+	assert.Equal(t, testBookTitle1, b1.Title)
+	assert.Equal(t, testBookAuthor1, b1.Author)
+	assert.Equal(t, playerID, b1.OwnerID)
+	assert.Equal(t, specificBookEntity, b1.Id)
+
+	b2 := BookTable.Filter(w, Book{
+		Title:  testBookTitle2,
+		Author: testBookAuthor2,
+	}, []string{"Title", "Author"})
+	require.Len(t, b2, 1)
+	assert.Equal(t, playerID, BookTable.Get(w, b2[0]).OwnerID)
+
+	b3 := BookTable.Filter(w, Book{
+		Title:  testBookTitle3,
+		Author: testBookAuthor3,
+	}, []string{"Title", "Author"})
+	require.Len(t, b3, 1)
+	assert.Equal(t, player2ID, BookTable.Get(w, b3[0]).OwnerID)
+}
+
+func tearDown(ws *websocket.Conn, server *http.Server) {
+	ws.Close()
+	server.Close()
+	time.Sleep(time.Second)
+}
+
+func TestUpdate(t *testing.T) {
+	e, ws, s, mockErrorHandler := startServer(t)
+	defer tearDown(ws, s)
+
+	w := e.World
+
+	playerID := 7
+
+	w.AddEntity()
```

Comment:
We should probably just start the entity manager from 1

Reply:
Yeah let me do this in a follow PR so not everything breaks lol